app/models/host.rb

```diff
@@ -1,18 +1,11 @@
 module Host
-  def self.method_missing(method, *args, &block)
-    type = "Host::Managed"
-    case method.to_s
-    when /create/, 'new'
-      if args.empty? || args[0].nil? # got no parameters
-        # set the default type
-        args = [{:type => type}]
-      else # got some parameters
-        args[0][:type] ||= type # adds the type if it doesn't exists
-        type = args[0][:type]   # stores the type for later usage.
-      end
-    end
+  def self.method_missing(method, *args, **kwargs, &block)
     if type.constantize.respond_to?(method, true)
```

Comment:
Now I see why this causes the infinite loop: `type` is not defined so that causes a method lookup.
```suggestion
    type = "Host::Managed"
    if type.constantize.respond_to?(method, true)
```